google/auth/impersonated_credentials.py

```diff
@@ -219,6 +219,8 @@ def __init__(
                 and self._source_credentials._always_use_jwt_access
             ):
                 self._source_credentials._create_self_signed_jwt(None)
+        if (self.universe_domain != self._source_credentials.universe_domain):
```

Comment:
maybe we should throw error if the universes are different?

Reply:
it is by design, that we don't set universe explicitly for impersonated credentials

We might want to throw if source_cred is GDU and impersonated cred somehow is not, but I think it is not possible

Reply:
it gets set to GDU automatically in base credential